### PR TITLE
earthly: init at 0.5.11

### PR DESCRIPTION
**Pull request by zoedsoupe**
###### Motivation for this change
Earthly can help to generate repeatable and understandable builds on CI or locally.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/earthly/default.nix
Comment (SuperSandro2000) [inline pkgs/development/tools/earthly/default.nix:14]: Why did you disable tests?

Comment (Mdsp9070) [reply pkgs/development/tools/earthly/default.nix]: Yeah, you're right, no reason at all.

**Comment by NobbZ** [comment]
Can this get in before 21.05 branch off?

I really need `earthly` for a customer project, and I managed to defer the rewrite of related stuff until next month with "earthly is not officially supported on my linux distro currently but should be with the next release in about a month".

I'd be happy if the PR gets at least merged into master/unstable by then, but 21.05 would be much more appreciated.

**Review by bryanasdev000 — APPROVED** [review]
LGTM, I did some quick tests, it looks like I have a new tool in my toolbox.
The only thing I think would help is a post-installation phase to rename `debugger` and` shellrepeater` to something like `earthly- *` to not collide with other random binaries.

**Comment by bryanasdev000** [comment]
@Mdsp9070 you need to add yourself to `maintainers/maintainer-list.nix` to fix errors of offborg.

Also, seems that `pkgs/top-level/all-packages.nix` have a Trailing whitespace at 2406.

**Comment by bryanasdev000** [comment]
> Can this get in before 21.05 branch off?
> 
> I really need `earthly` for a customer project, and I managed to defer the rewrite of related stuff until next month with "earthly is not officially supported on my linux distro currently but should be with the next release in about a month".
> 
> I'd be happy if the PR gets at least merged into master/unstable by then, but 21.05 would be much more appreciated.

I don't think it's a problem, when @Mdsp9070  fixes the question of maintainers we can ping here https://discourse.nixos.org/t/prs-already-reviewed/2617/375 or on discord to speed up the merge.

We have until May 21st according to https://discourse.nixos.org/t/21-05-release-schedule/12528 to 21.05 release.

**Comment by Mdsp9070** [comment]
> @Mdsp9070 you need to add yourself to `maintainers/maintainer-list.nix` to fix errors of offborg.
> 
> Also, seems that `pkgs/top-level/all-packages.nix` have a Trailing whitespace at 2406.

I added myself to `maintainers/maintainer-list.nix`; however I didn't find this trainling white space... I executed with nixpkgs-fmt but it changed so many lines so I gave up.

**Comment by Mdsp9070** [comment]
> LGTM, I did some quick tests, it looks like I have a new tool in my toolbox.
> The only thing I think would help is a post-installation phase to rename `debugger` and` shellrepeater` to something like `earthly- *` to not collide with other random binaries.

How I could make this? I'm pretty new to Nix derivations.

I tried:
```nix
postInstall = ''
  mv $src/cmd/debugger $src/cmd/earthly-debugger
  mv $src/cmd/shellrepeater $src/cmd/earthly-shellrepeater
'';
```

But this obviously fails with `Permisson denied`.

the `$out` directory isn't only for the binary output?

**Review by bryanasdev000 — APPROVED** [review]
LGTM, you will probably need to squash before merge.

File: pkgs/development/tools/earthly/default.nix
Comment (bryanasdev000) [inline pkgs/development/tools/earthly/default.nix:26]: This need to be `mdsp` the same as in `maintainer-list.nix`.

**Review by bryanasdev000 — COMMENTED** [review]
Maintainer name;

File: maintainers/maintainer-list.nix
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:11165]: Please put this into a separate commit with the message ``maintainers: add mdsp`` and sort it alphabetically.

File: pkgs/top-level/all-packages.nix
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:2377]:
```suggestion
  earthly = callPackage ../development/tools/earthly { };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request merged**
